pages/api/jam.js

```diff
@@ -1,33 +1,101 @@
 import fire from '../../config/firebaseAdminConfig';
 
+function getJamByUrlPath(jamUrlPath) {
+  const db = fire.firestore();
+  const jamsRef = db.collection('jams');
+
+  return jamsRef
+    .where('urlPath', '==', jamUrlPath)
+    .get()
+    .then((querySnapshot) => {
+      let jams = [];
+      querySnapshot.forEach((doc) => {
+        const jam = doc.data();
+        jam.key = doc.id;
+        jams.push(jam);
+      });
+      return jams[0];
+    });
+}
+
+function createJam({ name, description, statements }) {
+  const db = fire.firestore();
+  const jamsRef = db.collection('jams');
+  const batch = db.batch();
+
+  const randomString = (Math.random() + 1).toString(36).substring(2);
+
+  const jamId = jamsRef.doc();
+
+  batch.set(jamId, {
+    // adminId: '',
```

Comment:
Yeah. This will require a bit of fiddling with Auth0. 
Another thing we haven't captured yet is how to protect some of these API endpoints - only admins should be able to create Jams. This will require us sending an Auth0 JWT token to the function and validate access there -- will add a card for both these tasks on the board.